### PR TITLE
fix(indexer.api): Add msg hash to withdrawal response

### PR DESCRIPTION
**Pull request by ethenotethan**
**Description**
- Updated translation logic to populate cross-domain hash in `WithdrawalItem` when constructing `WithdrawalResponse`
- Moved translation logic to `models.go` and added a unit test

**Tests**
Added test which enforces that all `WithdrawalResponse` fields are appropriately populated after translation

Please describe any tests you've added. If you've added no tests, or left important behavior untested, please explain why not.

**Additional context**

Add any other context about the problem you're solving.

**Metadata**

- Fixes #[Link to Issue]

### CODE REVIEW

**Review by roninjin10 — APPROVED** [review]
Looks great ty!

File: indexer/api/models/models.go [hunk 1]
```diff
@@ -43,3 +48,38 @@ type WithdrawalResponse struct {
...
+
+// FIXME make a pure function that returns a struct instead of newWithdrawalResponse
+// newWithdrawalResponse ... Converts a database.L2BridgeWithdrawalsResponse to an api.WithdrawalResponse
+func CreateWithdrawalResponse(withdrawals *database.L2BridgeWithdrawalsResponse) WithdrawalResponse {
+	items := make([]WithdrawalItem, len(withdrawals.Withdrawals))
+	for i, withdrawal := range withdrawals.Withdrawals {
+
+		msg := withdrawal.L2BridgeWithdrawal.CrossDomainMessageHash
+		if msg == nil {
+			msg = &common.Hash{}
```
Comment (hamdiallam) [inline indexer/api/models/models.go]: this sgtm, the 0x0 can be an indicator if the withdrawal bypassed the cross domain messenger

File: indexer/api/models/models.go [hunk 2]
```diff
@@ -43,3 +48,38 @@ type WithdrawalResponse struct {
...
+
+		item := WithdrawalItem{
+			Guid:                 withdrawal.L2BridgeWithdrawal.TransactionWithdrawalHash.String(),
+			L2BlockHash:          withdrawal.L2BlockHash.String(),
+			Timestamp:            withdrawal.L2BridgeWithdrawal.Tx.Timestamp,
+			From:                 withdrawal.L2BridgeWithdrawal.Tx.FromAddress.String(),
+			To:                   withdrawal.L2BridgeWithdrawal.Tx.ToAddress.String(),
+			TransactionHash:      withdrawal.L2TransactionHash.String(),
+			Amount:               withdrawal.L2BridgeWithdrawal.Tx.Amount.String(),
+			MessageHash:          msg.String(),
```
Comment (hamdiallam) [inline indexer/api/models/models.go]: Should we name this `CrossDomainMessageHash` to be specific?

File: indexer/api/models/models.go [hunk 3]
```diff
@@ -43,3 +48,38 @@ type WithdrawalResponse struct {
...
+			ProofTransactionHash: withdrawal.ProvenL1TransactionHash.String(),
```
Comment (hamdiallam) [inline indexer/api/models/models.go]: I feel like we should update the Proof/Claim terminology to `Proven` and `Finalize`. Proof/Claim is very pre-bedrock

wdyt?

**Review by hamdiallam — APPROVED** [review]
Generally lgtm

File: indexer/api/models/models.go (hunk 2 above)
Comment (ethenotethan) [reply indexer/api/models/models.go]: Yee thats a great callout. If not mistaken wouldn't the GUID be the same thing as a withdrawal `MessageHash`? (assuming its referring to the `MessagePassed` event emitted by the `L2ToL1MessagePasser` contract)

File: indexer/api/models/models.go (hunk 3 above)
Comment (ethenotethan) [reply indexer/api/models/models.go]: 100%

**Review by hamdiallam — APPROVED** [review]

**Review by p1r5 — COMMENTED** [review]
epociask:indexer.api.add_withdrawal_hash

**Review by p1r5 — COMMENTED** [review]
indexer/api-ts/generated.ts

**Review by p1r5 — COMMENTED** [review]
epociask:indexer.api.add_withdrawal_hash

**Pull request merged**